src/types/route.ts

```diff
@@ -41,12 +45,12 @@ export type Route<
   /**
    * The specific route properties that were matched in the current route.
   */
-  matched: CreateRouteOptionsMatched<TName>,
+  matched: TMatches extends [...any[], infer Last extends CreatedRouteOptions] ? Last : CreatedRouteOptions,
```

Comment:
Nit: Like to see a utility type for this rather than having a conditional type inlined here.

Reply:
I added a `LastInArray` to @/types/utilities.ts, let me know what you think

Reply:
Yeah that looks nice, the fallback is for when we just use `Route` without any generics?